src/mlpack/methods/neighbor_search/neighbor_search.hpp

```diff
@@ -413,18 +500,21 @@ class NeighborSearch
   size_t Scores() const { return scores; }
 
   //! Access whether or not search is done in naive linear scan mode.
+  //! Deprecated. Will be removed in mlpack 3.0.0.
```

Comment:
Yes, I agree that it can be confussing. I have added some comments in: https://github.com/mlpack/mlpack/pull/765/commits/0d071af207002e9c2054bd60f85765942994e0ff